src/tests/ftest/osa/osa_offline_drain.py

```diff
@@ -0,0 +1,169 @@
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+
+    def run_offline_drain_test(self, num_pool, data=False):
+        """Run the offline drain without data.
+            Args:
+            num_pool (int) : total pools to create for testing purposes.
+            data (bool) : whether pool has no data or to create
+                          some data in pool. Defaults to False.
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        target_list = []
+        drain_servers = len(self.hostlist_servers) - 1
+
+        # Exclude target : random two targets
+        n = random.randint(1, 7)
+        target_list.append(n)
+        target_list.append(n+1)
+        t_string = "{},{}".format(target_list[0], target_list[1])
+
+        # Drain a rank (or server)
+        rank = random.randint(1, drain_servers)
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+            if data:
+                self.write_single_object()
+
+        # Drain the pool_uuid, rank and targets
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+            pver_begin = self.get_pool_version()
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+            output = self.dmg_command.pool_drain(self.pool.uuid,
+                                                 rank, t_string)
+            self.log.info(output)
+            time.sleep(10)
+            pver_drain = self.get_pool_version()
+            self.log.info("Pool Version after drain %d", pver_drain)
+            # Check pool version incremented after pool drain
+            self.assertTrue(pver_drain > pver_begin,
+                            "Pool Version Error:  After drain")
+
+        for val in range(0, num_pool):
+            display_string = "Pool{} space at the End".format(val)
+            pool[val].display_pool_daos_space(display_string)
+            pool[val].destroy()
+
+    def test_osa_offline_drain(self):
+        """Test ID: DAOS-4750
+        Test Description: Validate Offline Drain
+
+        :avocado: tags=all,pr,hw,large,osa,osa_drain,offline_drain
+        """
```

Comment:
Spacing is needed and a more descriptive description would be better.
```
"""
JIRA ID: DAOS-4750

Test Description: Validate Offline Drain

:avocado: tags=all,pr,hw,large,osa,osa_drain,offline_drain
"""
```

Reply:
Done.